src/main/java/com/google/sps/data/EventUpdater.java

```diff
@@ -36,7 +54,137 @@ public Entity getEntity() {
     return this.entity;
   }
 
-  public void updateEvent(HttpServletRequest request, GivrUser user, boolean forRegistration, EmbeddedEntity historyUpdate) throws IllegalArgumentException {
+  // Called when creating and updating Event from AddEventServlet and EditEventServlet.
+  public void updateEvent(HttpServletRequest request, GivrUser user, boolean forRegistration, EmbeddedEntity historyUpdate) throws IllegalArgumentException {    
+    // Must check if requesting User is a Moderator of the Event's Organization.
+    long ownerOrgId = 0;
+    try {
+      ownerOrgId = Long.parseLong(RequestHandler.getParameterOrThrow(request, "event-primary-organization-id"));
+    } catch (IllegalArgumentException err) {
+      logger.log(Level.SEVERE, "The primary organization ID is not valid.");
+      throw new IllegalArgumentException();
+    }
+
+    if (!user.isModeratorOfOrgWithId(ownerOrgId)) {
+      throw new IllegalArgumentException("Requesting user does not have the right credentials to create or update this Event.");
+    }
+
+    HashMap<String, String> formProperties = new HashMap<String, String>();
+
+    // Format is (Form EntryName, Entity PropertyName)
+    formProperties.put("event-primary-organization-id", "eventOwnerOrgId");
+    formProperties.put("event-name", "eventName");
+    formProperties.put("event-partner", "eventPartnerNames");
+    formProperties.put("event-details", "eventDetails");
+    formProperties.put("event-contact-email", "eventContactEmail");
+    formProperties.put("event-contact-phone-num", "eventContactPhone");
+    formProperties.put("event-contact-name", "eventContactName");
+    formProperties.put("event-street-address", "eventStreetAddress");
+    formProperties.put("event-city", "eventCity");
+    formProperties.put("event-state", "eventState");
+    formProperties.put("event-zip-code", "eventZipcode");
+
+    /* Optional Properties can be left blank in the request form.
+     *
+     * The following properties are optional:
+     * - eventPartnerNames
+     * - eventDetails
+     */
+    Set<String> optionalProperties = new HashSet<String>();
+    optionalProperties.add("eventPartnerNames");
+    optionalProperties.add("eventDetails");
+
+    for (Map.Entry<String, String> entry: formProperties.entrySet()) {
```

Comment:
I almost want to extract this into the request handler helper as well (this for loop i mean). You would have to pass in a lambda for the final set form value call. I'm imagining:
```
// pseudocode
ParserHelper.SetPropertiesFromForm(
      request,    
      formProperties,
      function(String propertyKey, String formValue) {
           setEventProperty(propertyKey, formValue);
      });
```

Reply:
Would this function also handle the other logic in the for loop, inside of the setPropertiesFromForm function?

Reply:
I'm going to add this into the Issue list, because there are a couple of PRs waiting for this!